models/public/nyu-fcrn-tf/nyu-fcrn-tf.md

```diff
@@ -0,0 +1,111 @@
+# nyu-fcrn-tf
+
+## Use Case and High-Level Description
+
+NYU-FCRN is a model for monocular depth estimation trained by NYU Depth V2 dataset,
```

Comment:
In original paper and repository I found only `ResNet50-UpProg` name for the network. Is better to use original name or proposed name is clearer?

Reply:
on my view, proposed name better, it is the same like trained checkpoint

Reply:
FCRN is only approach for solving depth estimation problem. But concrete model was called `ResNet-UpProj` in the article and repo by authors (they also have `ResNet-UpConv` model, also FCRN). And others refer to `ResNet-UpProj` in their papers (e.g. [here](http://fastdepth.mit.edu/2019_icra_fastdepth.pdf)).

Reply:
there is no any other pretrained and convertable to OV format.
the repo, checkpoint and module where model provided named FCRN, real name hide deep inside and does correlate with task (somebody, who does not read paper can think that it is classification model.
I do not agree to provide model with bit exact author name, I can accept only combination task and name.

Reply:
I see there's some bikeshedding going on, so I have to weigh in. 😉

First of all, `ResNet-UpProj` doesn't identify the network as a whole, but only a specific variant within the family of models described in the article. I skimmed the article, and it seems that the authors tried several different variants of their architecture and `ResNet-UpProj` was the one that performed the best. So we could include this ID as a qualifier (as `resnet-upproj`), but I don't think it should be the whole name.

The whole name should also identify the family itself, which, as far as I can see, doesn't have a concrete name, but I think it's reasonable to use the name of the repository as a stand-in. `fcrn` or `fcrn-dp` would be good choices. I think `fcrn` by itself is a bit iffy, because it's a bit too generic, so I would prefer the latter option.

We also need to identify the dataset, since the authors trained the model on multiple datasets. We already have a precedent for this with GMCNN, where we appended the dataset name as a qualifier. So I think we should do the same here and append `nyu-depth-v2` to the model name.

Finally, the model is available in multiple frameworks, so I guess we should append `tf`, even though it's kind of annoying.

Altogether, I think the model should be called `fcrn-dp-nyu-depth-v2-tf` or `fcrn-dp-resnet-upproj-nyu-depth-v2-tf`. The latter is a bit of a mouthful, and I don't think we _need_ to mention the variant, because the authors didn't upload any other variants; so I think I would prefer the former.

Reply:
@IRDonch can I remove dp since we have `depth` in the name (`fcrn-nyu-depth-v2-tf` )?

Reply:
I don't think that a good justification. That `depth` is a part of the dataset name, while "dp" comes from the repo name (and, coincidentally, clarifies the task type).